core/bin/snapshots_creator/src/main.rs

```diff
@@ -0,0 +1,251 @@
+        .put(l1_batch_number, &factory_deps)
+        .await
+        .context("Error storing factory deps in blob store")?;
+    let output_filepath_prefix = blob_store.get_storage_prefix::<SnapshotFactoryDependencies>();
+    let output_filepath = format!("{output_filepath_prefix}/{filename}");
+    tracing::info!(
+        "Finished processing factory dependencies, output stored in {}",
+        output_filepath
+    );
+    Ok(output_filepath)
+}
+
+async fn run(blob_store: Box<dyn ObjectStore>, pool: ConnectionPool) -> anyhow::Result<()> {
+    let mut conn = pool.access_storage().await?;
+    let start_time = seconds_since_epoch();
+
+    let l1_batch_number = conn.blocks_dal().get_sealed_l1_batch_number().await? - 1; // we subtract 1 so that after restore, EN node has at least one l1 batch to fetch
+
+    if conn
+        .snapshots_dal()
+        .get_snapshot_metadata(l1_batch_number)
+        .await?
+        .is_some()
+    {
+        tracing::info!(
+            "Snapshot for L1 batch number {} already exists, exiting",
+            l1_batch_number
+        );
+        return Ok(());
+    }
+
+    let miniblock_number = conn
+        .blocks_dal()
+        .get_miniblock_range_of_l1_batch(l1_batch_number)
+        .await?
+        .unwrap()
+        .1;
+    let storage_logs_count = conn
+        .snapshots_creator_dal()
+        .get_storage_logs_count(l1_batch_number)
+        .await?;
+
+    drop(conn);
+
+    //TODO load this from config
+    let chunk_size = 1_000_000;
+    let chunks_count = ceil_div(storage_logs_count, chunk_size);
+
+    tracing::info!(
+        "Creating snapshot for storage logs up to miniblock {}, l1_batch {}",
+        miniblock_number,
+        l1_batch_number.0
+    );
+    tracing::info!(
+        "Starting to generate {} chunks of max size {}",
+        chunks_count,
+        chunk_size
+    );
+
+    let factory_deps_output_file =
+        process_factory_deps(&*blob_store, &pool, miniblock_number, l1_batch_number).await?;
+
+    let mut storage_logs_output_files = vec![];
+
+    for chunk_id in 0..chunks_count {
+        tracing::info!(
+            "Processing storage logs chunk {}/{chunks_count}",
+            chunk_id + 1
+        );
+        let output_file = process_storage_logs_single_chunk(
+            &*blob_store,
+            &pool,
+            l1_batch_number,
+            chunk_id,
+            chunk_size,
+            chunks_count,
+        )
+        .await?;
+        storage_logs_output_files.push(output_file.clone());
+    }
+
+    let mut conn = pool.access_storage().await?;
+
+    conn.snapshots_dal()
+        .add_snapshot(
+            l1_batch_number,
+            &storage_logs_output_files,
+            factory_deps_output_file,
+        )
+        .await?;
+
+    METRICS.snapshot_l1_batch.set(l1_batch_number.0.as_u64());
+    METRICS.storage_logs_chunks_count.set(chunks_count);
+    METRICS
+        .snapshot_generation_timestamp
+        .set(seconds_since_epoch());
```

Comment:
Is this really a useful metric? IMO, other metrics you've defined make sense, but I struggle to understand what will this one allow to measure.

Reply:
I wanted it so that I could later define an alert on "last snapthot age"

Reply:
I wasn't aware it was possible to extract 'last update time' from prometheus, thanks! I removed this metric